archaius-zookeeper/src/main/java/com/netflix/config/source/ZooKeeperConfigurationSource.java

```diff
@@ -223,6 +224,10 @@ synchronized void deleteZkProperty(String key) throws Exception {
     }
     
     public void close() {
-    	Closeables.closeQuietly(pathChildrenCache);
+        try {
+            Closeables.close(pathChildrenCache, true);
+        } catch (IOException exc) {
+            logger.warn("IOException should not have been thrown.", exc);
```

Comment:
Thanks for the fix.  Happy to merge it when I see the new patch come by.

On Fri, Mar 6, 2015 at 7:33 PM, John <notifications@github.com> wrote:

> In
> archaius-zookeeper/src/main/java/com/netflix/config/source/ZooKeeperConfigurationSource.java
> <https://github.com/Netflix/archaius/pull/218#discussion_r25992847>:
>
> > @@ -223,6 +224,10 @@ synchronized void deleteZkProperty(String key) throws Exception {
> >      }
> >
> >      public void close() {
> > -    	Closeables.closeQuietly(pathChildrenCache);
> > +        try {
> > +            Closeables.close(pathChildrenCache, true);
> > +        } catch (IOException exc) {
> > +            logger.warn("IOException should not have been thrown.", exc);
>
> That's reasonable. I'll update this patch.
>
> Though I think the same behavior is inherently altered as it's going to
> SLF4J instead of JUL (and I'd be surprised if most users were importing a
> JUL-SLF4J bridge). But I'm sure nobody wants more JUL!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Netflix/archaius/pull/218/files#r25992847>.
>